frontend/lib/resolution/disambiguation.cpp

```diff
@@ -1031,10 +1031,234 @@ static void discardWorseWhereClauses(const DisambiguationContext& dctx,
   // TODO: fill me in
 }
 
+static bool isNegative(const Param* p) {
+  if (auto ip = p->toIntParam()) {
+    int64_t val = ip->value();
+    return val < 0;
+  }
+  return false;
+}
+
+static bool isNegativeParamToUnsigned(const Param* actualSym,
+                                      const Type* actualScalarType,
+                                      const Type* formalType) {
+  if (actualScalarType->isIntType() && formalType->isUintType()) {
+   return isNegative(actualSym);
+  }
+  return false;
+}
+
+static bool isMatchingImagComplex(Type* actualVt, Type* formalVt) {
+  if(auto fct = formalVt->toComplexType()) {
+    if (auto ait = actualVt->toImagType()) {
+      return ait->bitwidth()*2 == fct->bitwidth();
+    } else if (auto art = actualVt->toRealType()) {
+      return art->bitwidth()*2 == fct->bitwidth();
+    }
+  }
+  return false;
+}
+
+static void computeConversionInfo(const DisambiguationContext& dctx,
+                                  DisambiguationCandidate* candidate) {
+  // no need to recompute it if it is already computed
+  if (candidate->nImplicitConversionsComputed) {
+    return;
+  }
+
+  bool anyNegParamToUnsigned = false;
+  int numParamNarrowing = 0;
+  int nImplicitConversions = 0;
+
+  bool forGenericInit = candidate->fn->untyped()->name()==USTR("init") ||
+                        candidate->fn->untyped()->name()==USTR("init=");
+  size_t n = (size_t)dctx.call->numActuals();
+  for (size_t k = 0; k < n; k++) {
+
+    const FormalActual* fa1 = candidate->formalActualMap.byActualIdx(k);
+
+    if (fa1 == nullptr) {
+      // this can happen with some operators, or can it in Dyno?
+      continue;
+    }
+    if (forGenericInit && k < 2) {
+      // Initializer work-around: Skip _mt/_this for generic initializers
+      continue; // doubt we still need this as it should be handled by accessor
```

Comment:
We don't have "method token" in dyno, but we do have `this` formals. So AFAIK this should be `k < 1`.